includes/class-analytics.php

```diff
@@ -838,5 +839,41 @@ protected static function output_js_ini_load_event( $event ) {
 		</script>
 		<?php
 	}
+
+	/**
+	 * Add GA4 analytics support to AMP pages.
+	 */
+	public static function insert_ga4_analytics() {
+		if ( function_exists( 'is_amp_endpoint' ) && ! is_amp_endpoint() ) {
```

Comment:
```suggestion
		if ( ! function_exists( 'is_amp_endpoint' ) || ! is_amp_endpoint() ) {
```

If the function doesn't exist, should assume AMP is not installed and also return.

Reply:
This will have the same effect, if the function does not exist then the right-hand side of `&&` won't be evaluated. That's how it's done in [other](https://github.com/Automattic/newspack-plugin/blob/e89eda4e1b01f4799865b592b1bf9be14d0ee7e4/includes/class-nrh.php#L169) [places](https://github.com/Automattic/newspack-plugin/blob/e89eda4e1b01f4799865b592b1bf9be14d0ee7e4/includes/class-analytics.php#L562), so I'd prefer to keep it consistent.

Reply:
But there's a logic error.

This is fine when evaluating `function_exists( 'is_amp_endpoint' ) && is_amp_endpoint()` to return if AMP is available and active. To have the opposite effect you should either `! ( function_exists( 'is_amp_endpoint' ) && is_amp_endpoint() )` or `! function_exists( 'is_amp_endpoint' ) || ! is_amp_endpoint()`.